solutions/dmoj-Phidias.mdx

```diff
@@ -0,0 +1,75 @@
+---
+id: dmoj-phidias
+source: DMOJ
+title: Phidas
+author: Jesse Choe
+---
+
+Let $dp[i][j]$ be the minimum residual area with a rectangular slab size of $i X j$. Initialize $dp[i][j]$ as $i * j$ and set $dp[x_i][y_i]$ as 0, where $x_i, y_i$ are the width and heights of the desired plate sizes, respectively. Check every sub-rectangle with either $i$ or $j$ fixed (but not both) and store the residual area of the optimal sub-rectangle.
+
+**Time Complexity:** $\mathcal{O}(W * H * (W + H))$.
+
+```cpp
+#include <bits/stdc++.h>
+ 
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+using pi = pair<int, int>;
+ 
+#define f first
+#define s second
+
+/*
+ * Let dp[i][j] be the minimum residual area with a 
+ * rectangular slab size of i X j
+ */
+ 
+const int MAXN = 601;
```

Comment:
it'd be cool if you changed this to 600 and made it `MAX_N + 1` instead since that's the actual max value